File: frontend/src/app/services/profile.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse, ProfileUpdateDto, UserDto } from '../models/models';

const API_BASE = 'http://localhost:8080/api/v1';

@Injectable({ providedIn: 'root' })
export class ProfileService {
  constructor(private http: HttpClient) {}

  getProfile(): Observable<ApiResponse<UserDto>> {
    return this.http.get<ApiResponse<UserDto>>(`${API_BASE}/profile`);
  }

  updateProfile(dto: ProfileUpdateDto): Observable<ApiResponse<UserDto>> {
    return this.http.put<ApiResponse<UserDto>>(`${API_BASE}/profile`, dto);
  }

  updateProfileImage(imageUrl: string): Observable<ApiResponse<UserDto>> {
    const params = new HttpParams().set('imageUrl', imageUrl);
    return this.http.patch<ApiResponse<UserDto>>(`${API_BASE}/profile/image`, null, { params });
  }
}

File: backend/src/main/java/com/yaquedo/service/ProfileService.java
package com.yaquedo.service;

import com.yaquedo.dto.ApiResponse;
import com.yaquedo.dto.ProfileUpdateDto;
import com.yaquedo.dto.UserDto;

public interface ProfileService {

    ApiResponse<UserDto> getProfile(Long userId);

    ApiResponse<UserDto> updateProfile(Long userId, ProfileUpdateDto dto);

    ApiResponse<UserDto> updateProfileImage(Long userId, String imageUrl);
}
